bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/config/impl/HttpCacheConfigImpl.java

```diff
@@ -0,0 +1,131 @@
+                      + "be one of the cache stores active in this installation. Mandatory parameter.",
+              options = {
+                      @PropertyOption(name = HttpCacheStore.VALUE_MEM_CACHE_STORE_TYPE,
+                                         value = HttpCacheStore.VALUE_MEM_CACHE_STORE_TYPE),
+                      @PropertyOption(name = HttpCacheStore.VALUE_DISK_CACHE_STORE_TYPE,
+                                      value = HttpCacheStore.VALUE_DISK_CACHE_STORE_TYPE),
+                      @PropertyOption(name = HttpCacheStore.VALUE_JCR_CACHE_STORE_TYPE,
+                                      value = HttpCacheStore.VALUE_JCR_CACHE_STORE_TYPE)},
+              value = HttpCacheStore.VALUE_MEM_CACHE_STORE_TYPE)
+    // @formatter:on
+    private static final String PROP_CACHE_STORE = "httpcache.config.cachestore";
+    private static final String DEFAULT_CACHE_STORE = "MEM"; // Defaults to memory cache store
+    private String cacheStore;
+
+    @Property(label = "JCR path pattern (REGEX) for cache invalidation ",
+              description = "Optional set of paths in JCR (Oak) repository for which this cache has to be invalidated" +
+                      ". This accepts " + "REGEX. Example - /etc/my-products(.*)",
+              cardinality = Integer.MAX_VALUE)
+    private static final String PROP_CACHE_INVALIDATION_PATH_PATTERN = "httpcache.config.invalidation.oakpath";
+    private String[] cacheInvalidationPathPattern;
+
+    @Override
+    public String getRequestUri() {
+        return requestUriPattern;
+    }
+
+    @Override
+    public boolean isRequestAuthenticationRequired() {
+        return isRequestAuthenticationRequired;
+    }
+
+    @Override
+    public List<String> getUserGroupNames() {
+        return Arrays.asList(userGroups);
+    }
+
+    @Override
+    public String getCacheStoreName() {
+        return cacheStore;
+    }
+
+    @Override
+    public List<String> getCacheInvalidationPaths() {
+        return Arrays.asList(cacheInvalidationPathPattern);
+    }
+
+    @Activate
+    @Modified
+    protected void activate(Map<String, Object> configs) {
+        //Read configs and populate variables after trimming whitespaces.
+        requestUriPattern = PropertiesUtil.toString(configs.get(PROP_REQUEST_URI_PATTERN), StringUtils.EMPTY).trim();
+
+        isRequestAuthenticationRequired = PropertiesUtil.toBoolean(configs.get
+                (PROP_IS_REQUEST_AUTHENTICATION_REQUIRED), DEFAULT_IS_REQUEST_AUTHENTICATION_REQUIRED);
+
+        userGroups = PropertiesUtil.toStringArray(PROP_USER_GROUPS, ArrayUtils.EMPTY_STRING_ARRAY);
+        for (int i = 0; i < userGroups.length; i++) {
+            userGroups[i] = userGroups[i].trim();
+        }
+
+        cacheStore = PropertiesUtil.toString(configs.get(PROP_CACHE_STORE), StringUtils.EMPTY).trim();
+
+        cacheInvalidationPathPattern = PropertiesUtil.toStringArray(PROP_CACHE_INVALIDATION_PATH_PATTERN, ArrayUtils
```

Comment:
WDYT about compiling these into Patterns in activate; Compiling patterns is generally the expensive part of regex.